lib/administrate/field/select_basic.rb

```diff
@@ -8,23 +8,33 @@ class Engine < ::Rails::Engine
       end      
       
       def choices
-        options.fetch(:choices, []).
-          map { |o| prettify? ? [prettify(o), o] : o }
+        options.fetch(:choices, []).map { |o| convert(o) }
       end
   
       def to_s
-        prettify? ? prettify(data) : data
+        convert(data, raw: true)
       end
   
       def include_blank
         options.fetch(:include_blank, false)
       end
     
       private
+
+      def convert item, raw: false
+        return [prettify(item), item] if prettify? && !raw
+        return prettify(item) if prettify? && raw
+        return internationalise(item) if i18n?
+        return item
+      end
```

Comment:
I don't like this code at all - will return to it once I've made a bit more progress on my app